src/react_native/core.cljs

```diff
@@ -134,26 +134,46 @@
 
 (def use-context react/useContext)
 
+(defn use-ref-atom
+  [value]
+  (let [ref (use-ref (atom value))]
+    (.-current ^js ref)))
+
+(defn get-js-deps [deps]
+  (let [prev-deps (use-ref-atom {:prev-val false
+                                 :deps     []})]
+    (if deps
+      (if (not= deps (:deps @prev-deps))
+        (let [new-val (not (:prev-val @prev-deps))]
+          (reset! prev-deps {:prev-val new-val
+                             :deps     deps})
+          #js [new-val])
+        #js [(:prev-val @prev-deps)])
+      js/undefined)))
+
 (defn use-effect
```

Comment:
maybe should add a docstring to the hooks explaining what's happening with the deps or/and add smth about it in the guidelines/docs.